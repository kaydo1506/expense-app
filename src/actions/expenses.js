import { push, ref } from 'firebase/database';
import db from '../firebase/firebase';
// import uuid from 'uuid';
// what happens
// 1- component calls action generator
// 2- action generator returns object
// 3- component dispatches object
// 4- redux store changes

// what will hapen with firebase
// 1- components call action generator
// 2- action generator returns function
// 3- component dispatches function (?)
// 4- funtion runs (has the ability to dispatch other actions and do whatever it wants)

// ADD_EXPENSE action generator
export const addExpense = (expense) => ({
    type: 'ADD_EXPENSE',
    expense,
});
// addExpense starter
export const startAddExpense = (expenseData = {}) => {
    // middleware function
    return (dispatch) => {
        // destructuring
        const { description = '', note = '', amount = 0, createdAt = 0 } = expenseData;

        const expense = { description, note, amount, createdAt };

        return push(ref(db, 'expenses'), expense).then((ref) => {
            dispatch(
                addExpense({
                    id: ref.key,
                    ...expense,
                })
            );
        });
    };
};

// REMOVE_EXPENSE action generator
export const removeExpense = ({ id } = {}) => ({
    type: 'REMOVE_EXPENSE',
    id,
});
// EDIT_EXPENSE
export const editExpense = (id, updates) => ({
    type: 'EDIT_EXPENSE',
    id,
    updates,
});
