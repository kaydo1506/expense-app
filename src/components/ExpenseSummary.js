import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import getVisibleExpenses from '../selectors/expenses';
import expensesTotal from '../selectors/expenses-total';

export const ExpenseSummary = ({ expenses }) => {
    const res = expensesTotal(expenses);

    const ans = numeral(res / 100).format('0,0.00');
    return (
        <div>
            <h2>
                {' '}
                {expenses.length === 1
                    ? `Viewing ${expenses.length} expense totalling ₦${ans}`
                    : `Viewing ${expenses.length} expenses totalling ₦${ans}`}
            </h2>
        </div>
    );
};
const mapStateToProps = (state) => {
    return {
        expenses: getVisibleExpenses(state.expenses, state.filters),
    };
};
export default connect(mapStateToProps)(ExpenseSummary);
