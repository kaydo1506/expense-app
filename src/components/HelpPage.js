import React from 'react';

const HelpPage = () => <div>Get help here</div>;

export default HelpPage;
