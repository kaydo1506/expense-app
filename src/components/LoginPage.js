import React from 'react';

const LoginPage = ({ history }) => {
    const onSubmit = () => {
        history.push('/dashboard');
    };
    return (
        <div>
            <button onClick={onSubmit}>Login</button>
        </div>
    );
};

export default LoginPage;
