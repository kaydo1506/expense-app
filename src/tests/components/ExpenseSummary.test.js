import React from 'react';
import { ExpenseSummary } from '../../components/ExpenseSummary';
import expenses from '../fixtures/expenses';
import { shallow } from 'enzyme';

test('should render expense summary with single expense', () => {
    const wrapper = shallow(<ExpenseSummary expenses={[expenses[1]]} />);
    expect(wrapper).toMatchSnapshot();
});
test('should render expense summary with multiple expenses', () => {
    const wrapper = shallow(<ExpenseSummary expenses={expenses} />);
    expect(wrapper).toMatchSnapshot();
});
