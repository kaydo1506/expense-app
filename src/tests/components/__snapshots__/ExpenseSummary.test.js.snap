// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense summary with multiple expenses 1`] = `
<div>
  <h2>
     
    Viewing 3 expenses totalling ₦1,947.00
  </h2>
</div>
`;

exports[`should render expense summary with single expense 1`] = `
<div>
  <h2>
     
    Viewing 1 expense totalling ₦1,900.05
  </h2>
</div>
`;
